### DISCUSSION TITLE
Syncing and running autoSfM and SemiF-Annotation pipelines

### DISCUSSION THREAD

**Discussion by savadikarc** [precision-sustainable-ag/SemiF-AnnotationPipeline#11] [Ideas]
As the development stands now, the AutoSfm and SemiF-Annotation are two separate repositories, and it's probably a good thing to keep them that way. But this is making the the combined execution a bit confusing. How about we make this repo the main repo for handling the exection for the entire SemiF pipeline, and trigger the execution of the AutoSfM from here? This may be necessary to make sure the proper batch is processed. It can go as follows:

1. Consider ```autoSfM``` a separate package with it's own input and output requirements. Write some functionality here to copy data from Azure Blob storage to the VM (in the appropriate directory) and to copy the products from the autoSfM to another Blob storage.
2. Dynamically generate the config file required for the ```autoSfM ```, which needs a ```batch_id``` to run.
3. Trigger the autoSfM pipeline
4. Write some functionality to copy the autoSfM products from the Blob storage to the VM, and run SemiF-Annotation pipeline.

Let me know what you think @mkutu @skovsens !

**Comment by skovsens** [comment]
I think your suggestion makes a lot of sense @savadikarc and I agree completely.
If the integration with autoSfM is made first, we can reuse that for the other modules as well

**Comment by mkutu** [comment]
Same. Sounds good @savadikarc and thanks for being on top of this.
Is this something you can get started with?

**Reply by savadikarc** [reply]
Yep. I can start working on this.

**Comment by mkutu** [comment]
@savadikarc can you break this down a little more on Wednesday's meeting?

**Reply by savadikarc** [reply]
Sure thing.
